arbutil/wait_for_l1.go

```diff
@@ -126,75 +46,48 @@ func SendTxAsCall(ctx context.Context, client L1Interface, tx *types.Transaction
 	return client.CallContract(ctx, callMsg, blockNum)
 }
 
-func EnsureTxSucceededWithTimeout(ctx context.Context, client L1Interface, tx *types.Transaction, timeout time.Duration) (*types.Receipt, error) {
-	txRes, err := WaitForTx(ctx, client, tx.Hash(), timeout)
+func GetCallMsgBlockNumber(ctx context.Context, client L1Interface) (*big.Int, error) {
```

Comment:
Maybe name this "GetPendingCallBlockNumber"? You can do a call against non-pending state, but this specifically attempts to query the pending state.